### PR TITLE
DAOS-1357: Add new test for pool attributes.

### PR DESCRIPTION
**Pull request by skirvan44**
Signed-off-by: Scott Kirvan <scott.kirvan@intel.com>

### CODE REVIEW

File: src/tests/ftest/pool/Attribute.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+  test_machines:
+    - boro-26
```
Comment (ravalsam) [inline src/tests/ftest/pool/Attribute.yaml]: boro-A

Comment (ravalsam) [inline src/tests/ftest/pool/Attribute.yaml]: boro-A

File: src/tests/ftest/pool/Attribute.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  value_handles: !mux
+    validvalue:
+      value:
+        - "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyz"
+        - PASS
+    specialcharacter:
+      value:
+        - "! @#$%^&*()_+{}|:<>?-=[];',./"
+        - PASS
+#Defect DAOS-1330, enable after it resolved as server is getting Assert
```
Comment (ravalsam) [inline src/tests/ftest/pool/Attribute.yaml]: This defect has been closed so may be we should Enabled this value to see if it works or not

File: src/tests/ftest/pool/Attribute.yaml (hunk 1 above)
Comment (skirvan44) [reply src/tests/ftest/pool/Attribute.yaml]: Fixed

File: src/tests/ftest/pool/Attribute.yaml (hunk 2 above)
Comment (skirvan44) [reply src/tests/ftest/pool/Attribute.yaml]: Still doesn't work, opened new defect

**Comment by brianjmurrell** [comment]
@skirvan44 The checkpatch failure is addressed in daos-stack/code_review#9.  It seems that the GitHub user name can be empty (which was unexpected) and causing checkpatch to barf.

File: src/tests/ftest/pool/Attribute.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    """
+    verify the length of the Attribute names
+    """
+    # length of all the attribute names
+    aggregate_len = sum(len(attr_name) for attr_name in indata.keys()) + 1
+    # there is a space between each name, so account for that
+    aggregate_len += len(indata.keys())-1
+
+    if aggregate_len != size:
+        raise DaosApiError("FAIL: Size is not matching for Names in list"
```
Comment (sdwillso) [inline src/tests/ftest/pool/Attribute.py]: Is this intended to be DaosApiError or DaosTestError? Leaving one note here, usage of the two is mixed throughout. Looks fine besides this.

Comment (skirvan44) [reply src/tests/ftest/pool/Attribute.py]: Yes, should not be API error.

**Comment by ravalsam** [comment]
LGTM

**Pull request merged**